From 73ca09a4446f5845b74fbb4f6bacdf6c12761d7f Mon Sep 17 00:00:00 2001
From: john stultz <johnstul@us.ibm.com>
Date: Tue, 4 Aug 2009 16:30:52 -0700
Subject: [PATCH] rt: Fix build error with !CONFIG_PREEMPT_RT

commit 0e41458903a4809a4489e035c1ed71611824cbcf in tip.

Here's a small build fix to get !CONFIG_PREEMPT_RT +
CONFIG_PREEMPT_HARDIRQS to build. Not sure why this change was
introduced between 2.6.29-rt and 2.6.31-rt, but this sets it back.

Tested with and without CONFIG_PREEMPT_RT and with and without
CONFIG_PREEMPT_HARDIRQS.

Signed-off-by: John Stultz <johnstul@us.ibm.com>
Cc: Clark Williams <williams@redhat.com>
Cc: Steven Rostedt <srostedt@redhat.com>
Cc: Darren Hart <dvhltc@us.ibm.com>
LKML-Reference: <1249428652.3289.19.camel@localhost>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
---
 include/linux/bottom_half.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/include/linux/bottom_half.h b/include/linux/bottom_half.h
index 2e0e961..37dc356 100644
--- a/include/linux/bottom_half.h
+++ b/include/linux/bottom_half.h
@@ -1,7 +1,7 @@
 #ifndef _LINUX_BH_H
 #define _LINUX_BH_H
 
-#ifdef CONFIG_PREEMPT_RT
+#ifdef CONFIG_PREEMPT_HARDIRQS
 # define local_bh_disable()		do { } while (0)
 # define __local_bh_disable(ip)		do { } while (0)
 # define _local_bh_enable()		do { } while (0)
-- 
1.7.0.4

